### ISSUE TITLE
Add accessibility/WCAG2/ARIA support

### ISSUE THREAD

**Issue by ttaranov** [NG-ZORRO/ng-zorro-antd#4707]
### Reproduction link
[https://escvxj.run.stackblitz.io/](https://escvxj.run.stackblitz.io/)

### Steps to reproduce
almost all components documented on https://ng.ant.design/components/form/en have web accessibility issues, according to WCAG and ARIA standards:
https://www.w3.org/TR/WCAG21/
https://www.w3.org/TR/wai-aria-1.1/

### What is expected?
- add appropriate aria-* roles defined and managed appropriately on actions (e.g. on drop down expand/collapse), 
- input items should should have labels (or aria labels)
- keyboard navigation support
- WCAG 2.0 color schemes

### What is actually happening?
all required ARIA roles and labels are missing, keyboard and navigation support could be improved

| Environment | Info |
|---|---|
| ng-zorro-antd | 8.5.2 |
| Browser | any |

**Comment by wendellhu95** [comment]
Support of a11y is listed in our milestone #3667. We would like any help from the community. :heart:

**Comment by krokofant** [comment]
@ttaranov @wendellhu95 How much of this should be handled by Ng-Zorro and how much should be handle by the consumer?

Is it an issue specific to the docs? Consumers can add aria labels to their inputs and adjust their themes. I imagine some aria attributes are harder for the consumer to specify.

**Comment by wendellhu95** [comment]
@krokofant The React version could be a good reference. They already have `aria-*` supported.

**Comment by estellechvl** [comment]
would really love this to be implemented. Also focus trap on submenus.

**Comment by krokofant** [comment]
The above PR is stale and probably won't get any updates. If someone want's to pick it up it's encouraged.

**Comment by lsn-tomaszp** [comment]
It should be a priority nowadays to ensure a11y support in such a popular UI lib. The "accessible website" became standard and choosing between e.g. Angular Material or AntD is so much harder (with a huge advantage of Material) in case of lack of basic A11Y support in very good and complex components you've created. Please add support (or enhance/fix) for nzSelect, nzDropdown and other components to become THE BEST in Angular UI Lib World :) It's really tough to fix each component in ANTD lib with directive, working as "overlay" to deliver aria-attributes and required keyboard support.

**Comment by wojtek1150** [comment]
+1

**Comment by ButterflyCoder** [comment]
+1

**Comment by lppedd** [comment]
@wzhudev last time I worked with NG-ZORRO was back in early 2020, when it seemed like major contributors were from Alibaba.

Looking at the commits/PRs it seems the organization structure has changed a bit, is it correct? Or do you still maintain the role you had back then? I'm asking so that I can figure out how to approach the library, and if I have to allocate time for contributions - in case there isn't a lot of activity.

**Comment by Sina7312** [comment]
I wish we could have milestone for this at least :(

**Comment by Laffery** [comment]
[angular-eslint](https://github.com/angular-eslint/angular-eslint/blob/main/packages/eslint-plugin-template/README.md) now supports to do lint for template accessibility. 

I think we can add these eslint rules to check the whole library and do fix according to suggestions.
